lib60870-C/src/iec60870/cs104/cs104_connection.c

```diff
@@ -1154,10 +1154,14 @@ encodeIOA(CS104_Connection self, Frame frame, int ioa)
 void
 CS104_Connection_sendStartDT(CS104_Connection self)
 {
+    /* TODO: should add a return value? */
 #if (CONFIG_USE_SEMAPHORES == 1)
     Semaphore_wait(self->conStateLock);
 #endif /* (CONFIG_USE_SEMAPHORES == 1) */
 
+    if (!isRunning(self))
+        return;
+
```

Comment:
Exiting w/o calling `Semaphore_post` will leave the semaphore in an inconsistent state.

Reply:
I just realized that isRunning() has the same semaphore guard so looks like I can just move the check outside.

Reply:
Yes, that's what's done in other places, though that does create a TOCTOU bug :(